packages/cashier/src/hooks/__tests__/useWS.test.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { useWS as useWSShared } from '@deriv/shared';
+import useWS from '../useWS';
+import { TSocketCallTypes } from '../../types';
+
+jest.mock('@deriv/shared');
+
+const mockUseWSShared = useWSShared as jest.MockedFunction<typeof useWSShared>;
+
+const UseWSExample = <T extends keyof TSocketCallTypes>({ name, request }: { name: T; request: any }) => {
+    const WS = useWS(name);
+
+    return (
+        <>
+            <p data-testid={'isLoading'}>{WS.isLoading ? 'true' : 'false'}</p>
+            <p data-testid={'error'}>{WS.error ? JSON.stringify(WS.error) : 'undefined'}</p>
+            <p data-testid={'data'}>{WS.data ? JSON.stringify(WS.data) : 'undefined'}</p>
+            <button data-testid={'send'} onClick={() => WS.send(request)}>
```

Comment:
Can we use our own components here?
<Text>
<Button>

Reply:
@matin-binary This is a temporary solution to test hooks, After upgrading to React 18 we will switch to [@testing-library/react-hooks](https://www.npmjs.com/package/@testing-library/react-hooks) and we won’t need a component to test the hook anymore.